### PR TITLE
Gko 421 page cron validation

### PR DESCRIPTION
**Pull request by kamiiiel**
## Issue

https://gravitee.atlassian.net/browse/GKO-445

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/ConfigurationSerializationMapper.java [hunk 1]
```diff
@@ -52,6 +53,23 @@ default String serializeConfiguration(Object configuration) {
         }
     }
 
+    @Named("serializeConfigMap")
+    default Map<String, Object> serializeConfigMap(Object configuration) {
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/ConfigurationSerializationMapper.java]: maybe convert would be a better name as we don't actually 'serialise' here WDYT ? this way we could name it `convertConfig` and avoid the confusion with k8s `ConfigMap`.

also, just curious, why aren't we just performing a cast here ?

as a side note, also wondering why the Object mapper is not defined as a static field.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/ImportCRDUseCase.java [hunk 2]
```diff
@@ -431,13 +431,13 @@ private void deleteRemovedPages(Map<String, PageCRD> pages, String apiId) {
...
-    private void createOrUpdatePages(Map<String, PageCRD> pageCrds, String apiId, AuditInfo auditInfo) {
-        if (pageCrds == null || pageCrds.isEmpty()) {
+    private void createOrUpdatePages(Map<String, PageCRD> pageCRDs, String apiId, AuditInfo auditInfo) {
+        if (pageCRDs == null || pageCRDs.isEmpty()) {
             return;
         }
 
         var now = Date.from(TimeProvider.now().toInstant());
-        List<Page> pages = pageCrds.values().stream().map(this::initPageFromCRD).toList();
+        List<Page> pages = pageCRDs.values().stream().map(PageModelFactory::fromCRDSpec).toList();
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/use_case/ImportCRDUseCase.java]: As var is a bit everywhere in the new package structure, I would go with it for the sake of consistency.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/adapter/ApiCRDAdapter.java [hunk 3]
```diff
@@ -77,4 +88,20 @@ default Set<MemberCRD> mapMembers(ExportApiEntity definition) {
                 .collect(Collectors.toSet())
             : null;
     }
+
+    @Named("deserializeConfigMap")
+    default Map<String, Object> deserializeConfigMap(String configuration) {
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/adapter/ApiCRDAdapter.java]: same as above without the convert thing, I would just go to `deserializeConfig`

**Comment by a-cordier** [comment]
How about adding an [IT test](https://github.com/gravitee-io/gravitee-api-management/blob/21a1a2866b21b9ffaa781ddf079cf6b0cc7a324e/gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApisResourceTest.java#L479) with `dryRun` ? It would improve coverage drastically IMO

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/resources/crd/with-valid-resource-config.json [hunk 4]
```diff
@@ -0,0 +1,106 @@
+{
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/resources/crd/with-valid-resource-config.json]: isn't it more like `with-valid-page-config` and `with-invalid-page-config`?

**Review by ytvnr — APPROVED** [review]

**Pull request merged**